case_studies/dc2_multidetection/train_config.yaml

```diff
@@ -0,0 +1,169 @@
+    nll_gating:
+      _target_: bliss.encoder.variational_dist.SourcesGating
+  - _target_: bliss.encoder.variational_dist.LogNormalFactor
+    name: star_fluxes
+    dim: 6
+    sample_rearrange: "b ht wt d -> b ht wt 1 d"
+    nll_rearrange: "b ht wt 1 d -> b ht wt d"
+    nll_gating:
+      _target_: bliss.encoder.variational_dist.StarGating
+  - _target_: bliss.encoder.variational_dist.LogNormalFactor
+    name: galaxy_fluxes
+    dim: 6
+    sample_rearrange: "b ht wt d -> b ht wt 1 d"
+    nll_rearrange: "b ht wt 1 d -> b ht wt d"
+    nll_gating:
+      _target_: bliss.encoder.variational_dist.GalaxyGating
+  # - _target_: bliss.encoder.variational_dist.BivariateNormalFactor
+  #   name: shear
+  #   sample_rearrange: "b ht wt d -> b ht wt 1 d"
+  #   nll_rearrange: "b ht wt 1 d -> b ht wt d"
+  #   nll_gating:
+  #     _target_: case_studies.dc2_cataloging.utils.variational_dist.Cosmodc2Gating
+  - _target_: bliss.encoder.variational_dist.BivariateNormalFactor
+    name: ellipticity
+    sample_rearrange: "b ht wt d -> b ht wt 1 d"
+    nll_rearrange: "b ht wt 1 d -> b ht wt d"
+    nll_gating:
+      _target_: case_studies.dc2_cataloging.utils.variational_dist.Cosmodc2Gating
+
+my_metrics:
+  detection_performance:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+  detection_performance_star:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+    filter_list:
+      - _target_: bliss.encoder.metrics.SourceTypeFilter
+        filter_type: "star"
+  detection_performance_galaxy:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+    filter_list:
+      - _target_: bliss.encoder.metrics.SourceTypeFilter
+        filter_type: "galaxy"
+  detection_performance_one_source_mask:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+    filter_list:
+      - _target_: case_studies.dc2_cataloging.utils.metrics.SourceCountFilter
+        filter_source_count: "1m"
+  detection_performance_two_sources_mask:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+    filter_list:
+      - _target_: case_studies.dc2_cataloging.utils.metrics.SourceCountFilter
+        filter_source_count: "2m"
+  detection_performance_more_than_two_sources_mask:
+    _target_: bliss.encoder.metrics.DetectionPerformance
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
+    filter_list:
+      - _target_: case_studies.dc2_cataloging.utils.metrics.SourceCountFilter
+        filter_source_count: "2plus"
+  source_type_accuracy:
+    _target_: bliss.encoder.metrics.SourceTypeAccuracy
+    bin_cutoffs: [23.9, 24.1, 24.5, 24.9, 25.6]
+    bin_type: "njymag"
```

Comment:
I've been reading the SDSS using AB magnitude too. (See https://classic.sdss.org/dr7/algorithms/fluxcal.php) Given that, is njymag vs mag the correct distinction? I do agree that mag has different meanings for different surveys, because the surveys bands have different filter curves, but I'm not that's captured by the njymag/mag distinction.

Reply:
Perhaps bliss (the main package) should be agnostic to the units of flux. Then each case study would use whatever units are appropriate for the survey.

Reply:
I realize that this may change the code of others' case stuides. Maybe it's better to put it in a specific PR.